src/sentry/integrations/slack/notify_action.py

```diff
@@ -10,14 +10,19 @@
 
 from .utils import build_attachment
 
+MEMBER_PREFIX = '@'
+CHANNEL_PREFIX = '#'
+strip_channel_chars = ''.join([MEMBER_PREFIX, CHANNEL_PREFIX])
+
 
 class SlackNotifyServiceForm(forms.Form):
     team = forms.ChoiceField(choices=(), widget=forms.Select(
         attrs={'style': 'width:150px'},
     ))
     channel = forms.CharField(widget=forms.TextInput(
-        attrs={'placeholder': 'i.e #critical-errors'},
+        attrs={'placeholder': 'i.e #critical or @evan'},
     ))
+    channel_id = forms.HiddenInput()
```

Comment:
Have to store the channel ID to be able to send user messages, unlike the channel, which you can just use the channel name for (and it didn't even care if it had the `#` prefix).